### ISSUE TITLE
built-in Nuxt debug logging function

### ISSUE THREAD

**Issue by danielroe** [nuxt/nuxt#18701]
We can improve DX for developers by giving a custom function (that will be omitted from final build) but which can output not only the logged information but also the file and line-number (with a 'click-to-open' link direct from devtools).

Inspired by https://elixir-lang.org/getting-started/debugging.html#dbg

**Comment by madebyfabian** [comment]
Great idea, that would also make an integration for error loggers like Sentry way easier.

**Comment by harlan-zw** [comment]
inspo https://myray.app/

**Comment by Baroshem** [comment]
Awesome idea, I would totally use that in my projects!

**Comment by manniL** [comment]
> For logging, I think there is this Winston tool that I considered adding to Nuxt Security some time ago.
Not sure if Winston can do more than built-in [consola](https://github.com/unjs/consola/) can do rn though 🤔

**Comment by Baroshem** [comment]
Yes Alex, I just added it here for reference. When diving deeper into what this functionality should do, probably we would need something custom.

I am excited, this will improve DX by a mile

**Comment by riddla** [comment]
@jschroeter, we are using the logger provided by `@nuxt/kit` within our Nuxt modules. See https://nuxt.com/docs/api/kit/logging for details.

**Comment by Atulkumar112** [comment]
> > For logging, I think there is this Winston tool that I considered adding to Nuxt Security some time ago.
> 
> Not sure if Winston can do more than built-in [consola](https://github.com/unjs/consola/) can do rn though 🤔

I want to use consola but it given me plan white text, neither any styling nor timing is there while using in nuxt3.

**Comment by jschroeter** [comment]
> we are using the logger provided by `@nuxt/kit` within our Nuxt modules. See https://nuxt.com/docs/api/kit/logging for details.

I tried that too, but unfortunately it only works in the module context. We're looking for a solution which ideally works everywhere, meaning also in Nuxt plugins, Nitro routes etc.
